test/SymbolGtest.cpp

```diff
@@ -0,0 +1,105 @@
+////////////////////////////////////////////////////////////////////////////
+/**
+ *  @file   SymbolGtest.cpp
+ *  @date   Wed February 23 2022
+ *  @brief  Unit test suite for the Sample class
+ */
+////////////////////////////////////////////////////////////////////////////
+
+#include <limits>
+#include <sstream>
+
+#include<gtest/gtest.h>
+#include<gmock/gmock.h>
+#include "../inc/Symbol.h"
+
+using namespace testing;
+
+class SymbolFixture: public ::testing::Test
+{
+    protected:
+        calculator::Symbol<int> intSymbolA;
+        calculator::Symbol<int> intSymbolBBound;
+        calculator::Symbol<int> intSymbolCBound;
+        calculator::Symbol<int> intSymbolDBound;
+        calculator::Symbol<int> intSymbolEBound;
+        calculator::Symbol<int> intSymbolFBound;
+        calculator::Symbol<int> intSymbolZ;
+        
+        calculator::Symbol<double> doubleSymbolA;
+        calculator::Symbol<double> doubleSymbolBBound;
+        calculator::Symbol<double> doubleSymbolCBound;
+        calculator::Symbol<double> doubleSymbolDBound;
+        calculator::Symbol<double> doubleSymbolEBound;
+        calculator::Symbol<double> doubleSymbolFBound;
+        calculator::Symbol<double> doubleSymbolZ;
+
+        SymbolFixture(): 
+                intSymbolA{'a'},
+                intSymbolBBound{'B'}, 
+                intSymbolCBound{'c'},
+                intSymbolDBound{'d'},
+                intSymbolEBound{'e'},
+                intSymbolFBound{'f'},
+                intSymbolZ{'Z'},
+                doubleSymbolA{'A'},
+                doubleSymbolBBound{'b'},
+                doubleSymbolCBound{'C'},
+                doubleSymbolDBound{'d'},
+                doubleSymbolEBound{'e'},
+                doubleSymbolFBound{'f'},
+                doubleSymbolZ{'z'}
+        {
+            intSymbolBBound.setValue(-12);
+            intSymbolCBound.setValue(0);
+            intSymbolDBound.setValue(42);
+            intSymbolEBound.setValue(std::numeric_limits<int>::max());
+            intSymbolFBound.setValue(std::numeric_limits<int>::min());
+
+            doubleSymbolBBound.setValue(-123.45);
+            doubleSymbolCBound.setValue(0.0);
+            doubleSymbolDBound.setValue(42.42e10);
+            doubleSymbolEBound.setValue(std::numeric_limits<double>::max());
+            doubleSymbolFBound.setValue(std::numeric_limits<double>::min());
+        }
+};
+TEST_F(SymbolFixture, hasValue)
```

Comment:
Also also also, test names should have approximately this format, `WhenFunctionNameIsCalledWithArgument_ThenObjectIsReturnedWithValue`
So, there can be 4ish parts:

- Name of Function
- Arguments defined (optional)
- What is returned
- Any other expected side effects

Another example:
`WhenCreateWidgetIsCalledWithExampleEnum_ThenNewExampleWidgetIsReturnedAndWidgetCountIsIncreasedByOne`

Unit test, by definition, should be as descriptive as possible and should be as small as possible it is better to have 5 separate tests that test `int_min`, `-1`, `0`, `1`, `int_max` than to cram them all in one test.